Documentation/ApiOverview/RequestHandling/Typo3Request.rst

```diff
@@ -0,0 +1,70 @@
+.. include:: ../../Includes.txt
+
+.. highlight:: php
+
+.. _typo3-request:
+
+====================
+TYPO3 Request Object
+====================
+
+The TYPO3 request object is a PSR-7 based `ServerRequest` object containing a TYPO3-specific attribute object for normalized
+server parameters. These normalized parameter for instance resolve variables
+if the instance is behind a reverse proxy.
+
+.. important::
+
+   This substitutes :php:`GeneralUtility::getIndpEnv()`.
+
+The object is available from :php:`ServerRequestInterface $request` objects as an attribute.
+The request object is passed to controllers, example:
+
+.. code-block:: php
+
+    /** @var NormalizedParams $normalizedParams */
+    $normalizedParams = $request->getAttribute('normalizedParams');
+    $requestPort = $normalizedParams->getRequestPort();
+
+
+The request object is also available as a global variable in :php:`$GLOBALS['TYPO3_REQUEST']`. This is a workaround for
+the core which has to access the server parameters at places where $request is not available. So, while this object is
+globally available during any HTTP request, it is considered bad practice to use it if it is possible to access the
```

Comment:
I'd like to avoid the `it if it is` gallore:

> ... to use this global object if the request is accessible in another, official way.

Reply:
:-D Will fix it